PirateInterpreter/Values/Variable.cs

```diff
@@ -9,35 +9,12 @@ public class Variable : BaseValue, IValue
     public Variable(string value, ILogger logger, InterpreterFactory interpreterFactory): base(value, logger)
     {
         InterpreterFactory = interpreterFactory;
-        Value = GetValue(value);
-    }
-
-    public object GetValue(string value)
-    {
-        var resultNode = SymbolTable.Instance(Logger).Get(value);
-        if (resultNode is not IValueNode)
-        {
-
-            var interpreter = InterpreterFactory.GetInterpreter(resultNode, Logger);
-            var result = interpreter.VisitNode();
-            if (result.Count == 1)
-            {
-                return result[0].Value;
-            }
-            else
-            {
-                throw new Exception("Variable is not a single value");
-            }
-        }
-        else
-        {
-            var valueNode = (IValueNode)resultNode;
-            return valueNode.Value.Value;
-        }
+        Value = SymbolTable.Instance(Logger).Get(value).Value;
```

Comment:
The SymbolTalbe.Instance...shouldn't need a Logger parameter every time....Think of a solution to make a Singleton with a logger without ask for it at the Instance method